DocLinkValidity.py

```diff
@@ -0,0 +1,59 @@
+import os
+import sys
+import json
+import re
+from urllib.request import Request, urlopen
+import urllib.error
+
+from utils.RuleReader import RuleReader
+
+SERVICE_FOLDER_PATH = os.getcwd() + '/services'
+
+if __name__ == "__main__":
+    attrName = 'ref'
+    rr = RuleReader(SERVICE_FOLDER_PATH)
+    
+    
+    ruleList = rr.getRulesAttr(attrName)
+    invalidRefDict = {
+        'NoRef': [],
+        'InvalidSyntax': []
+    }
+    
+    for rule in ruleList:
+        for ref in ruleList[rule][attrName]:
+            if ref.strip() == '':
+                invalidRefDict['NoRef'].append(rule)
+                continue
+            output = re.search(r'<(.*)>', ref)
+            if output is None:
+                invalidRefDict['InvalidSyntax'].append(rule)
+                continue
+            try:
+                url = output.group(1)
+                if url.strip() == '':
+                    invalidRefDict['NoRef'].append(rule)
+                    continue
+                
+                print('.', end="", flush=True)
+                conn = Request(url)
+                conn.add_header('User-Agent', 'aws-cli')
+                resp = urlopen(conn)
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

Audit url open for permitted schemes. Allowing use of file:/ or custom schemes is often unexpected. https://bandit.readthedocs.io/en/latest/blacklists/blacklist_calls.html#b310-urllib-urlopen